pennylane/operation.py

```diff
@@ -1046,6 +1077,43 @@ def __rmatmul__(self, other):
 
     __imatmul__ = __matmul__
 
+    def _data(self):
+
+        obs = self.non_identity_obs
+        tensor = set()
+
+        for ob in obs:
+            parameters = tuple(param.tostring() for param in ob.parameters)
+            tensor.add((ob.name, ob.wires, parameters))
+
+        return tensor
+
+    def __eq__(self, other):
+
+        if isinstance(other, (Tensor, Observable)):
+            val = self._data() == other._data()
+        if isinstance(other, qml.Hamiltonian):
+            val = other._data() == self._data()
+
+        return val
+
+    def __add__(self, other):
+
+        if isinstance(other, (Tensor, Observable)):
+            val = qml.Hamiltonian([1, 1], [self, other])
+        if isinstance(other, qml.Hamiltonian):
+            val = other + self
+
+        return val
+
+    def __mul__(self, a):
+        return qml.Hamiltonian([a], [self])
+
+    __rmul__ = __mul__
+
+    def __sub__(self, other):
+        return self.__add__(other.__mul__(-1))
```

Comment:
`Tensor` inherits from `Observable`, so these should be inherited; if they are no different, they can probably be deleted here!